Mathlib/RingTheory/MvPowerSeries/Evaluation.lean

```diff
@@ -0,0 +1,368 @@
+* `Continuous.tendsto_apply_variables_zero_of_cofinite`:
+  when `a s` tends to  zero for the filter of cofinite subsets of `σ`.
+
+* `MvPowerSeries.eval₂_domain` : the `Prop`-valued structure
+that is required to evaluate power series
+
+* `MvPowerSeries.uniformContinuous_eval₂` : uniform continuity of the evaluation
+
+* `MvPowerSeries.continuous_eval₂` : continuity of the evaluation
+
+* `MvPowerSeries.aeval` : the evaluation map as an algebra map
+
+-/
+
+namespace MvPowerSeries
+
+open Topology
+
+open Filter MvPolynomial RingHom Set TopologicalSpace UniformSpace
+
+/- ## Necessary conditions -/
+
+section
+
+variable {σ : Type*}
+variable {R : Type*} [CommRing R] [TopologicalSpace R]
+variable {S : Type*} [CommRing S] [TopologicalSpace S]
+variable {φ : R →+* S}
+
+-- We endow MvPowerSeries σ R with the Pi topology
+open WithPiTopology
+
+/-- Families at which power series can be evaluated -/
+structure EvalDomain (a : σ → S) : Prop where
+  hpow : ∀ s, IsTopologicallyNilpotent (a s)
+  tendsto_zero : Tendsto a cofinite (𝓝 0)
+
+/-- The domain of evaluation of `MvPowerSeries`, as an ideal -/
+def evalDomainIdeal [IsTopologicalRing S] [IsLinearTopology S S] : Ideal (σ → S) where
+  carrier := {a | EvalDomain a}
+  add_mem' {a} {b} ha hb := {
+    hpow := fun s ↦ IsTopologicallyNilpotent.add (ha.hpow s) (hb.hpow s)
+    tendsto_zero := by
+      rw [← add_zero 0]
+      exact ha.tendsto_zero.add hb.tendsto_zero }
+  zero_mem' := {
+    hpow := fun s ↦ by
+      simp only [Pi.zero_apply]
+      apply tendsto_atTop_of_eventually_const (i₀ := 1)
+      intro i hi
+      rw [zero_pow (Nat.ne_zero_iff_zero_lt.mpr hi)]
+    tendsto_zero := tendsto_const_nhds }
+  smul_mem' c {x} hx := {
+    hpow := fun s ↦ by
+      simp only [IsTopologicallyNilpotent, Pi.smul_apply', smul_eq_mul, mul_pow]
+      exact IsLinearTopology.tendsto_mul_zero_of_right _ _ (hx.hpow s)
+    tendsto_zero := IsLinearTopology.tendsto_mul_zero_of_right _ _ hx.tendsto_zero }
+
+theorem EvalDomain.add [IsTopologicalRing S] [IsLinearTopology S S]
+    {a b : σ → S} (ha : EvalDomain a) (hb : EvalDomain b) : EvalDomain (a + b) :=
+  evalDomainIdeal.add_mem' ha hb
+
+theorem EvalDomain.zero [IsTopologicalRing S] [IsLinearTopology S S] : EvalDomain (0 : σ → S) :=
+  evalDomainIdeal.zero_mem'
+
+theorem EvalDomain.smul_mem [IsTopologicalRing S] [IsLinearTopology S S]
+    (c : σ → S) {x : σ → S} (hx : EvalDomain x) : EvalDomain (c • x) :=
+  evalDomainIdeal.smul_mem' c hx
+
+theorem EvalDomain.comp {a : σ → R} (ha : EvalDomain a) {ε : R →+* S} (hε : Continuous ε) :
+    EvalDomain (ε ∘ a) := by
+  apply EvalDomain.mk _ ((Continuous.tendsto' hε 0 0 (map_zero ε)).comp ha.tendsto_zero)
+  · intro s
+    unfold IsTopologicallyNilpotent
+    convert (Continuous.tendsto' hε 0 (ε 0) rfl).comp (ha.hpow s)
+    · ext n; simp only [Function.comp_apply, map_pow]
+    · rw [map_zero]
+
+/-- Bourbaki, Algèbre, chap. 4, §4, n°3, Prop. 4 (i) (a & b) -/
```

Comment:
Could you make this into a clickable citation? Something like `[Bourbaki, *Algèbre*, chap. 4, §4, n°3, Prop. 4 (i) (a & b)](bourbaki1981)`. 

Note that `bourbaki1981` is the translation, so ideally it would be nice to either
- add another entry for the French version
- adapt your citation to the English version